nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateCsv.java

```diff
@@ -265,9 +281,31 @@ public CsvPreference getPreference(final ProcessContext context, final FlowFile
         // input is transferred over to Java as is. So when you type the characters "\"
         // and "n" into the UI the Java string will end up being those two characters
         // not the interpreted value "\n".
-        final String msgDemarcator = context.getProperty(END_OF_LINE_CHARACTER).evaluateAttributeExpressions(flowFile).getValue().replace("\\n", "\n").replace("\\r", "\r").replace("\\t", "\t");
-        return new CsvPreference.Builder(context.getProperty(QUOTE_CHARACTER).evaluateAttributeExpressions(flowFile).getValue().charAt(0),
-                context.getProperty(DELIMITER_CHARACTER).evaluateAttributeExpressions(flowFile).getValue().charAt(0), msgDemarcator).build();
+        final String msgDemarcator = context.getProperty(END_OF_LINE_CHARACTER)
+                .evaluateAttributeExpressions(flowFile)
+                .getValue()
+                .replace("\\n", "\n")
+                .replace("\\r", "\r")
+                .replace("\\t", "\t");
+
+        final char quoteChar = context.getProperty(QUOTE_CHARACTER)
+                .evaluateAttributeExpressions(flowFile)
+                .getValue()
+                .charAt(0);
+
+        final int delimiterChar = context.getProperty(DELIMITER_CHARACTER)
+                .evaluateAttributeExpressions(flowFile)
+                .getValue()
+                .charAt(0);
+
+        int maxLinesPerRow = 0;
+        if (context.getProperty(MAX_LINES_PER_ROW).isSet()) {
+            maxLinesPerRow = context.getProperty(MAX_LINES_PER_ROW).asInteger();
+        }
```

Comment:
The default value is 0 so you so can simplify this to
```suggestion
        int maxLinesPerRow =  context.getProperty(MAX_LINES_PER_ROW).asInteger();
```

Reply:
Property is now required with default 0 value.